src/lib.rs

```diff
@@ -0,0 +1,550 @@
+/// # }
+/// ```
+#[cfg(compute_heavy_executor_tokio)]
+pub fn initialize_block_in_place_strategy() {
+    log::info!("initializing compute-heavy executor with block in place strategy");
+    let strategy = ExecutorStrategy::BlockInPlace(BlockInPlaceExecutor {});
+    COMPUTE_HEAVY_FUTURE_EXECUTOR_STRATEGY
+        .set(strategy)
+        .unwrap_or_else(|_| panic!("Compute heavy future executor can only be initialized once"))
+}
+
+/// Creates a secondary tokio runtime to delegate compute-heavy futures to via channels.
+/// Uses the same number of threads as cpu core, and reduced thread priority on linux (niceness 10).
+/// Use secondary_executor_config() to customize the defaults.
+///
+/// # Panics
+/// Panics if compute-heavy executor strategy is initialized more than once, across all strategies. 
+/// 
+/// # Example
+/// 
+/// ```
+/// use compute_heavy_future_executor::initialize_secondary_tokio_runtime_strategy;
+/// use compute_heavy_future_executor::spawn_compute_heavy_future;
+/// 
+/// # async fn run() {
+/// initialize_secondary_tokio_runtime_strategy();
+/// 
+/// let future = async { 
+///     std::thread::sleep(std::time::Duration::from_millis(50));
+///     5
+///  }; 
+/// 
+/// let res = spawn_compute_heavy_future(future).await.unwrap();
+/// assert_eq!(res, 5);
+/// # }
+/// ```
+#[cfg(compute_heavy_executor_tokio)]
+pub fn initialize_secondary_tokio_runtime_strategy() {
+    log::info!("initializing compute-heavy executor with secondary tokio runtime strategy");
+    let strategy =
+        ExecutorStrategy::SecondaryTokioRuntime(SecondaryTokioRuntimeExecutor::new(None, None));
+    COMPUTE_HEAVY_FUTURE_EXECUTOR_STRATEGY
+        .set(strategy)
+        .unwrap_or_else(|_| panic!("Compute heavy future executor can only be initialized once"))
+}
+
+/// Creates a secondary tokio runtime to delegate compute-heavy futures to via channels.
+/// 
+/// Allows custom thread niceness of -20..=19 , and any thread count. If either value is omitted,
+/// it will fall back to the default of 10 niceness and/or count of threads = cpu core count.
+///
+/// # Panics
+/// Panics if niceness provided outside of -20..=19 bounds.
+/// 
+/// Panics if compute-heavy executor strategy is initialized more than once, across all strategies. 
+/// 
+/// # Example
+/// 
+/// ```
+/// use compute_heavy_future_executor::initialize_secondary_tokio_runtime_strategy_and_config;
+/// use compute_heavy_future_executor::spawn_compute_heavy_future;
+/// 
+/// # async fn run() {
+/// initialize_secondary_tokio_runtime_strategy_and_config(Some(5), Some(10));
+/// 
+/// let future = async { 
+///     std::thread::sleep(std::time::Duration::from_millis(50));
+///     5
+///  }; 
+/// 
+/// let res = spawn_compute_heavy_future(future).await.unwrap();
+/// assert_eq!(res, 5);
+/// # }
+/// ```
+#[cfg(compute_heavy_executor_tokio)]
+pub fn initialize_secondary_tokio_runtime_strategy_and_config(
+    niceness: Option<i8>,
+    thread_count: Option<usize>,
```

Comment:
Note: also add a customizable queue size for the background job queue, also concurrency limit on running tasks.